pyomo/solvers/plugins/solvers/SCIPAMPL.py

```diff
@@ -374,9 +369,11 @@ def _postsolve(self):
             if len(results.solution) > 0:
                 results.solution(0).status = SolutionStatus.optimal
             try:
-                if results.problem.sense == ProblemSense.minimize:
+                if results.solver.primal_bound < results.solver.dual_bound:
                     results.problem.lower_bound = results.solver.primal_bound
+                    results.problem.upper_bound = results.solver.dual_bound
                 else:
+                    results.problem.lower_bound = results.solver.dual_bound
                     results.problem.upper_bound = results.solver.primal_bound
```

Comment:
Can this go wrong if your bounds cross by a little bit?

Reply:
Yes - you probably can.  This will be easily fixable in the new solver interfaces (where the results parser can have easier access to the original model and won't have to guess about the sense), but there isn't an obvious better solution here (unless the scip output log prints something about the sense - but I really don't want to put more dependence on the parsing of output logs).